genesyscloud/routing_utilization/genesyscloud_routing_utilization_proxy.go

```diff
@@ -0,0 +1,99 @@
+type updateRoutingUtilizationFunc func(ctx context.Context, p *routingUtilizationProxy, request *platformclientv2.Utilizationrequest) (*platformclientv2.Utilizationresponse, *platformclientv2.APIResponse, error)
+type deleteRoutingUtilizationFunc func(ctx context.Context, p *routingUtilizationProxy) (*platformclientv2.APIResponse, error)
+
+type updateDirectlyFunc func(ctx context.Context, p *routingUtilizationProxy, d *schema.ResourceData, utilizationRequest []interface{}) (*platformclientv2.APIResponse, error)
+
+type routingUtilizationProxy struct {
+	clientConfig                 *platformclientv2.Configuration
+	routingApi                   *platformclientv2.RoutingApi
+	getRoutingUtilizationAttr    getRoutingUtilizationFunc
+	updateRoutingUtilizationAttr updateRoutingUtilizationFunc
+	deleteRoutingUtilizationAttr deleteRoutingUtilizationFunc
+
+	updateDirectlyAttr updateDirectlyFunc
+}
+
+func newRoutingUtilizationProxy(clientConfig *platformclientv2.Configuration) *routingUtilizationProxy {
+	api := platformclientv2.NewRoutingApiWithConfig(clientConfig)
+	return &routingUtilizationProxy{
+		clientConfig:                 clientConfig,
+		routingApi:                   api,
+		getRoutingUtilizationAttr:    getRoutingUtilizationFn,
+		updateRoutingUtilizationAttr: updateRoutingUtilizationFn,
+		deleteRoutingUtilizationAttr: deleteRoutingUtilizationFn,
+
+		updateDirectlyAttr: updateDirectlyFn,
+	}
+}
+
+func getRoutingUtilizationProxy(clientConfig *platformclientv2.Configuration) *routingUtilizationProxy {
+	if internalProxy == nil {
+		internalProxy = newRoutingUtilizationProxy(clientConfig)
+	}
+	return internalProxy
+}
+
+func (p *routingUtilizationProxy) getRoutingUtilization(ctx context.Context) (*platformclientv2.APIResponse, error) {
+	return p.getRoutingUtilizationAttr(ctx, p)
+}
+func (p *routingUtilizationProxy) updateRoutingUtilization(ctx context.Context, request *platformclientv2.Utilizationrequest) (*platformclientv2.Utilizationresponse, *platformclientv2.APIResponse, error) {
+	return p.updateRoutingUtilizationAttr(ctx, p, request)
+}
+func (p *routingUtilizationProxy) deleteRoutingUtilization(ctx context.Context) (*platformclientv2.APIResponse, error) {
+	return p.deleteRoutingUtilizationAttr(ctx, p)
+}
+
+func (p *routingUtilizationProxy) updateDirectly(ctx context.Context, d *schema.ResourceData, utilizationRequest []interface{}) (*platformclientv2.APIResponse, error) {
+	return p.updateDirectlyAttr(ctx, p, d, utilizationRequest)
+}
+
+// Calling the Utilization API directly while the label feature is not available.
+// Once it is, this code can go back to using platformclientv2's RoutingApi to make the call.
+func getRoutingUtilizationFn(ctx context.Context, p *routingUtilizationProxy) (*platformclientv2.APIResponse, error) {
+	apiClient := &p.routingApi.Configuration.APIClient
+	path := fmt.Sprintf("%s/api/v2/routing/utilization", p.routingApi.Configuration.BasePath)
```

Comment:
Just checked the API explorer, but it looks like this guy now has the endpoint available publicly.  I would create a separate ticket in Jira and assign it to Sprint 18 to go through and refactor all of this code to use the routing/utilization SDK calls instead of the API calls.

Reply:
Ticket Created and added to sprint 18